File: 1190-reverse-substrings-between-each-pair-of-parentheses/1190-reverse-substrings-between-each-pair-of-parentheses.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseParentheses = function(s) 
{
    let ans="",tmp="";
    let stack = [];
    for(let idx in s)
    {
        let x = s[idx];
        if(x!=')')
        {
            if(x=='(' || stack.length)
                stack.push(x);                
            else
                ans+=x;
        }
        else
        {
            tmp="";
            while(stack.length && stack[stack.length-1]!='(')
            {
                tmp+=stack.pop();
            }
            stack.pop();
            if(stack.length)
            {
                for(let y in tmp)
                    stack.push(tmp[y]);
            }
            else
                ans+=tmp;
        }
    }
    
    return ans;
};

File: 1190-reverse-substrings-between-each-pair-of-parentheses/1190-reverse-substrings-between-each-pair-of-parentheses.cpp
class Solution {
public:
    string reverseParentheses(string s) 
    {
       
        string ans="" , tmp;
        int n = s.size();
        stack<char>st;
        for(auto &x: s)
        {
            if(x!=')')
            {
                if(x=='(' || st.size())
                    st.push(x);
                else
                    ans+=x;
            }
            else
            {
                tmp="";
                while(st.size() and st.top()!='(')
                {
                    tmp+=st.top();
                    st.pop();
                }
                st.pop();
                if(st.size())
                {
                    for(auto &y: tmp)
                        st.push(y);
                }
                else
                    ans+=tmp;
            }
        }
        
        
        return ans;
        
        
        
    }
};